### DISCUSSION TITLE
[Art competition] Abstract Interpretation

### DISCUSSION THREAD

**Discussion by PSuHyeon** [prosyslab-classroom/cs524-program-analysis#269] [Art competition]
![DALL·E 2022-11-25 11 15 01 - in pastel style art, inside the leaves of a golden apple tree, a doctor robot is drawing the X-ray illustration of a golden apple ](https://user-images.githubusercontent.com/99390261/203887184-e508b06a-8fae-44ca-a468-64122e6f1568.png)

## Short Description
For me, [Abstract interpretation] means [In pastel style art, inside the leaves of a golden apple tree, a doctor robot is drawing the X-ray illustration of a golden apple ]".

## Long Description (up to 3 sentences)
For me, program is a golden apple that is very helpful but potentially have bug inside. In this context, abstract interpretation is taking x-ray image of golden apple to see 'abstract image' of the whole apple without taking a bite (i.e. single program execution). While in some cases, x-ray image might clearly depicts shape of bugs, in other cases, it might be produce ambiguous result to differentiate between bugs and seeds inside, and we say 'don't know'.

**Comment by KihongHeo** [comment]
Do you mean apple in mythology? or in princess Snow White?

**Comment by PSuHyeon** [comment]
Apple in princess snow white might be good example! With this robot, we don't need kiss of Prince Charming anymore🤣
